### PR TITLE
Add tests for the API contracts

### PR DESCRIPTION
**Pull request by jzoldak**
- Verify that the swagger docs are well formed
- Generate a stub server and start it up
- Test the API contracts against the running stub server
- Organize make targets for common commands

@efagin ready for review for merging
@benpatterson @estute FYI

### CODE REVIEW

**Comment by efagin** [comment]
this is really cool.

File: tests/test_oauth.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+- config:
+    - testset: 'Oauth API'
+
+- test:
+    - name: 'Get request returns a 405'
+    - url: '/oauth2/v1/access_token'
+    - method: 'GET'
+    - expected_status: [405]
```
Comment (jzoldak) [inline tests/test_oauth.yaml]: @efagin are we expecting a 405? that's what the stub server is responding with.

File: Makefile [hunk 2]
```diff
@@ -1,33 +1,53 @@
...
-	cd scripts/aws && python -m pytest
+clean.stub:
+	@rm -rf $(STUB_SERVER_FOLDER)
 
-validate_python:
-	make clean
-	make quality
-	make test_scripts
+# Download the swagger codegen jar if you don't already have it.
+# TODO: verify via checksum that the file is valid.
```
Comment (jzoldak) [inline Makefile]: @efagin Still thinking through how best to do the checksum that will work on either OSX or ubuntu. Open to suggestions!

Comment (efagin) [inline Makefile]: No idea! Just leave the TODO in there for now...

File: tests/test_oauth.yaml (hunk 1 above)
Comment (efagin) [inline tests/test_oauth.yaml]: Yes, that stub implementation returns a 405 when the method is not defined in the Swagger document (there's no `GET` defined for that endpoint!)

**Comment by efagin** [comment]
Preference: remove the `@` prefixes. More logging seems helpful here.

**Comment by efagin** [comment]
@jzoldak feedback requested on my makefile update - travis seems happy with the cleanup. general 👍 on your test suite.

**Pull request merged**